src/sentry/auth/helper.py

```diff
@@ -492,7 +491,7 @@ def _handle_unknown_identity(self, identity):
             # the email matches we go ahead and let them merge it. This is the
             # only way to prevent them having duplicate accounts, and because
             # we trust identity providers, its considered safe.
-            if existing_user and existing_user.is_managed and verified_email:
```

Comment:
so the behavior is changed a bit here, and assuming tests dont fail immediately I might need to update them (forgot about that before pushing)